src/hooks/useWeather.jsx

```diff
@@ -0,0 +1,27 @@
+import { useState, useEffect } from "react";
+import getWeather from "../api/weather";
+
+export const useWeather = (location) => {
+  const [weatherData, setWeatherData] = useState(null);
+  const [error, setError] = useState(null);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const fetchWeather = async () => {
+    try {
+      const weather = await getWeather(location);
+      setWeatherData(weather);
+    } catch (err) {
+      setError(err.message);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchWeather();
+  }, [location]);
+
+  return [weatherData, fetchWeather, isLoading, error];
```

Comment:
Very nice hook!
The only thing I'd change here is for the hook to return an object instead of an array.
That's because, if it happened that `error` is all that you need, you'd need to do:

```
const [, , , error] = useWeather();
```

while you could just do:

```
const { error } = useWeather()
```

because objects don't care about the order.